WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -133,15 +133,24 @@ public async Task<bool> StartCoinJoinAsync(IEnumerable<SmartCoin> coinCandidates
 
 	/// <summary>Attempt to participate in a specified round.</summary>
 	/// <param name="roundState">Defines the round parameter and state information to use.</param>
-	/// <returns>Whether or not the round resulted in a successful transaction.</returns>
+	/// <returns><c>True</c>: client should end the current CoinJoin progress. <c>False</c>: continue with the blame rounds.</returns>
 	public async Task<bool> StartRoundAsync(IEnumerable<SmartCoin> smartCoins, RoundState roundState, CancellationToken cancellationToken)
 	{
 		var constructionState = roundState.Assert<ConstructionState>();
 
 		// Register coins.
 		using PersonCircuit personCircuit = HttpClientFactory.NewHttpClientWithPersonCircuit(out Tor.Http.IHttpClient httpClient);
+		ImmutableArray<AliceClient> registeredAliceClients = ImmutableArray<AliceClient>.Empty;
+		try
+		{
+			registeredAliceClients = await CreateRegisterAndConfirmCoinsAsync(httpClient, smartCoins, roundState, cancellationToken).ConfigureAwait(false);
+		}
+		catch (InvalidOperationException ex)
```

Comment:
This isn't specific enough exception to catch IMO. Can you make it more specific?

Reply:
ACK